### PR TITLE
Configurable MongoDB or Postgres Persistance

### PR DESCRIPTION
**Pull request by stevensJourney**
# Overview

This builds off https://github.com/powersync-ja/powersync-nodejs-backend-todolist-demo/pull/5 which adds MongoDB support. 

This PR updates the `data` APIs to use either MongoDB or Postgres as the backend database.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,7 +28,9 @@ The endpoints are as follows:
 
 ## Packages
 
-[node-postgres](https://github.com/brianc/node-postgres) is used to interact with the Postgres database when a PowerSync enabled client performs requests to the `/api/data` endpoint.
+[node-postgres](https://github.com/brianc/node-postgres) is used to interact with the Postgres database when a Postgres + PowerSync enabled client performs requests to the `/api/data` endpoint.
+
+[mongodb](https://www.npmjs.com/package/mongodb) iis used to interact with the Postgres database when a MongoDB + PowerSync enabled client performs requests to the `/api/data` endpoint.
```
Comment (rkistner) [inline README.md]:
```suggestion
[mongodb](https://www.npmjs.com/package/mongodb) is used to interact with the MongoDB database when a client performs requests to the `/api/data` endpoint.
```

File: README.md [hunk 2]
```diff
@@ -28,7 +28,9 @@ The endpoints are as follows:
 
 ## Packages
 
-[node-postgres](https://github.com/brianc/node-postgres) is used to interact with the Postgres database when a PowerSync enabled client performs requests to the `/api/data` endpoint.
+[node-postgres](https://github.com/brianc/node-postgres) is used to interact with the Postgres database when a Postgres + PowerSync enabled client performs requests to the `/api/data` endpoint.
```
Comment (rkistner) [inline README.md]:
```suggestion
[node-postgres](https://github.com/brianc/node-postgres) is used to interact with the Postgres database when a client performs requests to the `/api/data` endpoint.
```

The client itself is not specific to MongoDB or Postgres, so I'd call it just a "client" or "PowerSync-enabled client"

File: src/api/data.js [hunk 3]
```diff
@@ -1,24 +1,14 @@
...
-});
+const persistenceFactory = factories[config.database.type];
 
-pool.on('error', (err, client) => {
-  console.error('Pool connection failure to postgres:', err, client);
-});
+const { updateBatch } = await persistenceFactory(config.database.uri);
 
-/**
+/**s
```
Comment (rkistner) [inline src/api/data.js]:
```suggestion
/**
```

**Review by rkistner — DISMISSED** [review]
Added some copy suggestions, but happy with the code.

File: README.md (hunk 2 above)
Comment (stevensJourney) [reply README.md]: Thanks! This one was very silly.

**Review by rkistner — APPROVED** [review]

**Pull request merged**